<launch>
	<arg name="world" default="$(find prometheus_gazebo)/worlds/drl_worlds/drl_world_2.world"/>
    <!-- Rviz config -->
    <arg name="visualization" default="true"/>
    <arg name="rivz_config" default="$(find prometheus_drl)/launch/rviz_uav_test.rviz"/>

	<arg name="swarm_num" default="3"/>
    <arg name="agent_prefix" default="/uav"/>
    <!-- 0是离散控制,1是连续控制 -->
    <arg name="action_mode" default="0"/>
    <!-- 启动 fake_odom_node -->
    <node pkg="prometheus_drl" type="drl_main_node" name="drl_main_node" output="screen">
        <param name="swarm_num" value="$(arg swarm_num)"/>
        <param name="agent_prefix" value="$(arg agent_prefix)"/>
        <param name="action_mode" value="$(arg action_mode)"/>
        <param name="sim_mode" value="true" type="bool"/>
        <param name="block_size" value="0.5"/>
        <param name="sim_mode" value="true" type="bool"/>
        <!-- 地图参数 -->
        <param name="map/origin_x" value="-10.0" type="double"/>
        <param name="map/origin_y" value="-10.0" type="double"/>
        <param name="map/map_size_x" value="20.0" type="double"/>
        <param name="map/map_size_y" value="20.0" type="double"/>
        <param name="map/resolution" value="0.1" type="double"/>
        <param name="map/inflate" value="0.1" type="double"/>
        <param name="map/odom_inflate" value="0.0" type="double"/>
        <param name="map/queue_size" value="1" type="int"/>
        <param name="map/border" value="true" type="bool"/>
    </node>

    <node pkg="prometheus_drl" type="read_agent_state" name="read_agent_state" output="screen" launch-prefix="gnome-terminal --">
        <param name="agent_prefix" value="$(arg agent_prefix)"/>
    </node>

    <!-- 启动Gazebo -->
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(arg world)"/>
    </include>

	<!-- 启动Rviz-->
	<group if="$(arg visualization)">
        <node type="rviz" name="rviz" pkg="rviz" args="-d $(arg rivz_config)"/>
    </group>

	<node pkg="prometheus_drl" type="terminal_control" name="terminal_control" output="screen" launch-prefix="gnome-terminal --">	
        <param name="agent_prefix" value="$(arg agent_prefix)"/>
        <param name="action_mode" value="$(arg action_mode)"/>
    </node>	

</launch>

